plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -439,6 +440,21 @@ def set_kritis_breakglass
       template.dig_fetch(:metadata, :annotations)[:"kritis.grafeas.io/breakglass"] = "true"
     end
 
+    def set_istio_sidecar_injection
+      return unless Samson::EnvCheck.set?('ISTIO_INJECTION_SUPPORTED')
+      return unless @doc.deploy_group_role.inject_istio_annotation?
```

Comment:
`next unless template.dig(:spec, :template, :metadata)` so we don't set it for configmaps etc ... or only set it for `pod_template` which will then also support Pod and CronJob

Reply:
I actually had that line in there, and a test for it.  But I couldn't get that line to be triggered in my tests, so it was complaining about being an uncovered line.

Then I realized that the place I was calling this method up top is in a block that is only called for resources that have some variation of a pod template.  So the `set_istio_sidecar_injection` method wouldn't ever get called for a Service or ConfigMap.

Reply:
should still set it on the `pod_template` to reuse that logic / support Pod+CronJob